include/rocksdb/file_system.h

```diff
@@ -108,6 +108,20 @@ struct IOOptions {
   // such as NewRandomAccessFile and NewWritableFile.
   std::unordered_map<std::string, std::string> property_bag;
 
+  // Sets the property in property_bag.
+  void SetProperty(const std::string& property, const std::string& value) {
+    property_bag[property] = value;
+  }
+
+  // Returns the value of the property in the propery_bag. If property doesn't
+  // exist, it returns "Not found".
+  std::string GetProperty(const std::string& property) {
+    if (property_bag.find(property) == property_bag.end()) {
+      return "Not found";
```

Comment:
IMHO it's more typical to want to treat the empty string the same as unset, like is common practice for environment variables.

Also, if returning `const std::string&`, I suggest
```
  static const std::string kNotFound = "";
  return kNotFound;
```
to avoid any issues with the storage duration of a temporary std::string initialized from a const char[].